RFCs/rfc-0010-wallet-provider-attestation-api.md

```diff
@@ -0,0 +1,252 @@
+
+When a requester seeks attestation from a wallet, they must provide contextual bindings. These bindings are key-value pairs where the keys are strings, and the values can be any type with valid JSON encoding.
+
+#### `mina_requestAttestation` Binding Parameters
+
+#### `program`
+A unique identifier or hash representing the provable program the wallet should run internally for attestation. This might correspond to the verification key of the provable program or its hash.
+
+### `clientData`
+An adaptable data structure designed to bind the wallet to contextually relevant information. The content and structure of `clientData` are dynamic, tailored to the specific scenario or attestation requirement at hand. It can encapsulate a range of data, from verifiable credentials to custom datasets, based on what the wallet is expected to know when attesting. Given the myriad potential use cases, the shape and constituents of clientData can vary widely. Requesters should structure it based on prior knowledge of the data the wallet is acquainted with, ensuring effective communication and validation during the attestation process.
+
+#### `publicInputs`
+A key-value map detailing the provable-program's method signature for open inputs. Each key-value pair describes a specific input and the value to use in the provable-program; further binding the wallet.
+
+
+#### Example On-Chain Usage -- request attestation using a provable program for royalties claim
+```ts
+import { Mina } from "o1js";
+
+const mina = window.mina;
+
+try {
+    // Define the binding for the attestation
+    const binding = {
+        // The program could be constrained by a unique identifier for the program like a verification key
+        program: "ProvableProgramForRoyaltyClaim",
+        clientData: {
+            "type": ["VerifiableCredential", "ArtOwnershipCredential"],
+            "credentialData": {
+                "issuer": "did:mina:B62sj...owg",
+                "credentialSubject": {
+                    "id": "did:mina:B62rf...kf9",
+                }
+            }
+        },
+        // Define the required public inputs
+        publicInputs: {
+            "claim": {
+                "property": "credentialData.credentialSubject.royaltyPercentage",
+                "operation": "eq",
+                "value": "5"
+            },
+            "nonce": {
+                "value": "1"
+            }
+        },
+    };
+
+    // Request attestation using the binding. Wallet will match and verify internally against stored credentials.
+    const attestationProof = await mina.request({
+        method: 'mina_requestAttestation',
+        params: binding
+    });
+
+    // Construct a transaction using the received attestation proof
+    let transaction = await Mina.transaction(() => {
+        new RoyaltiesZkApp(zkappAddress).claim(attestationProof);
```

Comment:
Would claim be a function on the zkApp, that takes in a Proof type? (like [here](https://docs.minaprotocol.com/zkapps/o1js/recursion)?)

Reply:
Yes! It would be a method on the contract, or possibly another provable-program!